# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

In [2]:
https://lucid.app/lucidchart/36c47fee-eefb-4009-a469-d4eb57501f13/edit?invitationId=inv_505b5420-255e-40b4-a258-ddd3d86acec8


: Msg 102, Level 15, State 1, Line 1
Incorrect syntax near '/'.

Total execution time: 00:00:00.346

2. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 10pt; white-space: pre-wrap;">What is the datatype for women employees?</span>

In [9]:
SELECT * 
FROM LaborStatisticsDB.dbo.datatype
WHERE data_type_text = 'Women employees'


(1 row affected)

Total execution time: 00:00:00.074

data_type_code data_type_text 10 WOMEN EMPLOYEES

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector?</span>

In [18]:
SELECT * 
FROM LaborStatisticsDB.dbo.series
WHERE series_title = 'Women employees'
AND industry_code = '55522110'

SELECT industry_code, industry_name 
FROM LaborStatisticsDB.dbo.industry
WHERE industry_name LIKE '%commercial_bank%'

SELECT * 
FROM LaborStatisticsDB.dbo.supersector
WHERE supersector_name LIKE 'Financial activities'

(2 rows affected)

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.116

series_id supersector_code industry_code data_type_code seasonal series_title CES5552211010 55 55522110 10 S Women employees CEU5552211010 55 55522110 10 U Women employees

industry_code industry_name 55522110 Commercial banking

supersector_code supersector_name 55 Financial activities

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [9]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.annual_2016 as a                                                                                                                                                                                   
LEFT JOIN LaborStatisticsDB.dbo.series as s 
ON a.series_id = s.series_id

(100 rows affected)

Total execution time: 00:00:00.318

id series_id year period value footnote_codes original_file series_id supersector_code industry_code data_type_code seasonal series_title 8755 CEU0000000001 2016 M13 144306 NULL ce.data.0.AllCESSeries.csv CEU0000000001 0 00000000 1 U All employees 22220 CEU0000000001 2016 M13 144306 NULL ce.data.00a.TotalNonfarm.Employment.csv CEU0000000001 0 00000000 1 U All employees 22221 CEU0000000010 2016 M13 71492 NULL ce.data.00a.TotalNonfarm.Employment.csv CEU0000000010 0 00000000 10 U Women employees 8756 CEU0000000010 2016 M13 71492 NULL ce.data.0.AllCESSeries.csv CEU0000000010 0 00000000 10 U Women employees 8757 CEU0500000001 2016 M13 122083 NULL ce.data.0.AllCESSeries.csv CEU0500000001 5 05000000 1 U All employees 6281 CEU0500000001 2016 M13 122083 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000001 5 05000000 1 U All employees 7954 CEU0500000002 2016 M13 34.4 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000002 5 05000000 2 U Average weekly hours of all employees 8758 CEU0500000002 2016 M13 34.4 NULL ce.data.0.AllCESSeries.csv CEU0500000002 5 05000000 2 U Average weekly hours of all employees 8759 CEU0500000003 2016 M13 25.65 NULL ce.data.0.AllCESSeries.csv CEU0500000003 5 05000000 3 U Average hourly earnings of all employees 7955 CEU0500000003 2016 M13 25.65 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000003 5 05000000 3 U Average hourly earnings of all employees 6282 CEU0500000006 2016 M13 100531 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000006 5 05000000 6 U Production and nonsupervisory employees 8760 CEU0500000006 2016 M13 100531 NULL ce.data.0.AllCESSeries.csv CEU0500000006 5 05000000 6 U Production and nonsupervisory employees 8761 CEU0500000007 2016 M13 33.6 NULL ce.data.0.AllCESSeries.csv CEU0500000007 5 05000000 7 U Average weekly hours of production and nonsupervisory employees 5721 CEU0500000007 2016 M13 33.6 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv CEU0500000007 5 05000000 7 U Average weekly hours of production and nonsupervisory employees 5722 CEU0500000008 2016 M13 21.54 NULL ce.data.05c.TotalPrivate.ProductionEmployeeHoursAndEarnings.csv CEU0500000008 5 05000000 8 U Average hourly earnings of production and nonsupervisory employees 8762 CEU0500000008 2016 M13 21.54 NULL ce.data.0.AllCESSeries.csv CEU0500000008 5 05000000 8 U Average hourly earnings of production and nonsupervisory employees 8763 CEU0500000010 2016 M13 58747 NULL ce.data.0.AllCESSeries.csv CEU0500000010 5 05000000 10 U Women employees 6283 CEU0500000010 2016 M13 58747 NULL ce.data.05a.TotalPrivate.Employment.csv CEU0500000010 5 05000000 10 U Women employees 7956 CEU0500000011 2016 M13 881.55 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000011 5 05000000 11 U Average weekly earnings of all employees 8764 CEU0500000011 2016 M13 881.55 NULL ce.data.0.AllCESSeries.csv CEU0500000011 5 05000000 11 U Average weekly earnings of all employees 8765 CEU0500000012 2016 M13 367.3 NULL ce.data.0.AllCESSeries.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 7957 CEU0500000012 2016 M13 367.3 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 3808 CEU0500000012 2016 M13 367.3 NULL ce.data.02b.AllRealEarningsAE.csv CEU0500000012 5 05000000 12 U Average weekly earnings of all employees 3809 CEU0500000013 2016 M13 10.69 NULL ce.data.02b.AllRealEarningsAE.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 7958 CEU0500000013 2016 M13 10.69 NULL ce.data.05b.TotalPrivate.AllEmployeeHoursAndEarnings.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 8766 CEU0500000013 2016 M13 10.69 NULL ce.data.0.AllCESSeries.csv CEU0500000013 5 05000000 13 U Average hourly earnings of all employees 8767 CEU0500000016 2016 M13 105.3 NULL ce.data.0.AllCESSeries.csv CEU0500000016 5 05000000 16 U Indexes of aggregate weekly

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [10]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.series as s                                                                                                                                                                                   
JOIN LaborStatisticsDB.dbo.datatype as dt 
ON s.data_type_code = dt.data_type_code

(100 rows affected)

Total execution time: 00:00:00.386

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 10 WOMEN EMPLOYEES CES0000000025 0 00000000 25 S All employees 25 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 26 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 6 PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 7 AVERAGE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 8 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000010 5 05000000 10 S Women employees 10 WOMEN EMPLOYEES CES0500000011 5 05000000 11 S Average weekly earnings of all employees 11 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000012 5 05000000 12 S Average weekly earnings of all employees 12 AVERAGE WEEKLY EARNINGS OF ALL EMPLOYEES CES0500000013 5 05000000 13 S Average hourly earnings of all employees 13 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 16 INDEXES OF AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 17 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES CES0500000019 5 05000000 19 S Average weekly hours of all employees 19 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0500000021 5 05000000 21 S Diffusion indexes 21 DIFFUSION INDEXES CES0500000022 5 05000000 22 S Diffusion indexes 22 DIFFUSION INDEXES CES0500000023 5 05000000 23 S Diffusion indexes 23 DIFFUSION INDEXES CES0500000025 5 05000000 25 S All employees 25 ALL EMPLOYEES CES0500000026 5 05000000 26 S All employees 26 ALL EMPLOYEES CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 30 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 31 AVERAGE WEEKLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 32 AVERAGE HOURLY EARNINGS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 34 INDEXES OF AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 35 INDEXES OF AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000036 5 05000000 36 S Average weekly hours 36 AVERAGE WEEKLY HOURS CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 56 AGGREGATE WEEKLY HOURS OF ALL EMPLOYEES CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 57 AGGREGATE WEEKLY PAYROLLS OF ALL EMPLOYEES CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 81 AGGREGATE WEEKLY HOURS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 82 AGGREGATE WEEKLY PAYROLLS OF PRODUCTION AND NONSUPERVISORY EMPLOYEES CES0600000001 6 06000000 1 S All employees 1 ALL EMPLOYEES CES0600000002 6 06000000 2 S Average weekly hours of all employees 2 AVERAGE WEEKLY HOURS OF ALL EMPLOYEES CES0600000003 6 06000000 3 S Average hourly earnings of all employees 3 AVERAGE HOURLY EARNINGS OF ALL EMPLOYEES CES06

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [11]:
SELECT TOP 100 *
FROM LaborStatisticsDB.dbo.series as s                                                                                                                                                                                   
JOIN LaborStatisticsDB.dbo.industry as i 
ON s.industry_code = i.industry_code

(100 rows affected)

Total execution time: 00:00:00.299

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000010 5 05000000 10 S Women employees 1 5000000 - A Total private 1 T 2 CES0500000011 5 05000000 11 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000012 5 05000000 12 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000013 5 05000000 13 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000019 5 05000000 19 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000021 5 05000000 21 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000022 5 05000000 22 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000023 5 05000000 23 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000025 5 05000000 25 S All employees 1 5000000 - A Total private 1 T 2 CES0500000026 5 05000000 26 S All employees 1 5000000 - A Total private 1 T 2 CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000036 5 05000000 36 S Average weekly hours 1 5000000 - A Total private 1 T 2 CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0600000001 6 06000000 1 S All employees 2 6000000 - A Goods-producing 1 T 3 CES0600000002 6 06000000 2 S Average weekly hours of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000003 6 06000000 3 S Average hourly earnings of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000006 6 06000000 6 S Production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 CES0600000007 6 06000000 7 S Average weekly hours of production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 C

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.</span>

In [14]:
-- SELECT SUM(id)
-- FROM LaborStatisticsDB.dbo.industry
-- WHERE data_type_text = 'ALL EMPLOYEES'


SELECT ROUND(SUM(a.VALUE), 0) AS 'EMPLOYEES ACROSS INDUSTRIES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '01'

(1 row affected)

Total execution time: 00:00:00.109

EMPLOYEES ACROSS INDUSTRIES 2340612

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp;</span>

In [16]:
SELECT ROUND(SUM(a.VALUE), 0) AS 'WOMEN ACROSS INDUSTRIES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '10'
-- WHERE data_type_text = 'Women employees'

(1 row affected)

Total execution time: 00:00:00.113

WOMEN ACROSS INDUSTRIES 1125490

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;</span>

In [17]:
SELECT ROUND(SUM(a.VALUE), 0) AS 'PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.annual_2016 AS a 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = a.series_id
WHERE data_type_code = '6'

--  data_type_text = 'PRODUCTION AND NONSUPERVISORY EMPLOYEES'

(1 row affected)

Total execution time: 00:00:00.113

PRODUCTION/NONSUPERVISORY EMPLOYEES 1263650

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries?</span>

In [45]:
SELECT ROUND(AVG(j.value), 0) AS 'HOURS PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
WHERE  j.series_id LIKE '%07'


(1 row affected)

Total execution time: 00:00:00.178

HOURS PRODUCTION/NONSUPERVISORY EMPLOYEES 36

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please.</span>

In [46]:
SELECT ROUND(SUM(j.value), 0) AS 'PAYROLL PRODUCTION/NONSUPERVISORY EMPLOYEES'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
WHERE  j.series_id LIKE '%82'

(1 row affected)

Total execution time: 00:00:00.192

PAYROLL PRODUCTION/NONSUPERVISORY EMPLOYEES 1838753220

6. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [18]:
SELECT i.industry_name, ROUND(MAX(j.value), 0) AS 'HIGHESTHOURS'
FROM LaborStatisticsDB.dbo.january_2017 as j
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE j.series_id LIKE '%07'
GROUP BY i.industry_name
ORDER BY HIGHESTHOURS 

SELECT i.industry_name, ROUND(MIN(j.value), 0) AS 'LOWESTHOURS'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON s.series_id = j.series_id 
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code 
WHERE  j.series_id LIKE '%07'
GROUP BY industry_name
ORDER BY LOWESTHOURS

-- SELECT industry_name 
-- GROUP BY 


(501 rows affected)

(501 rows affected)

Total execution time: 00:00:00.292

industry_name HIGHESTHOURS Fitness and recreational sports centers 17 Civic and social organizations 19 Clothing stores 20 Bowling centers 20 Family clothing stores 20 Skiing facilities 20 Women's clothing stores 20 Drinking places 21 Clothing accessories stores 22 Clothing and clothing accessories stores 22 Amusements 22 Arts 23 Snack and nonalcoholic beverage bars 23 Other personal care services 23 Other home furnishings stores 24 Shoe stores 24 Restaurants and other eating places 24 Book stores and news dealers 24 Food services and drinking places 24 Gift 24 Full-service restaurants 25 Golf courses and country clubs 25 Leisure and hospitality 25 Limited-service restaurants 25 Museums 25 Accommodation and food services 25 Performing arts companies 25 Personal care services 25 Pet care services 26 Performing arts and spectator sports 26 Office supplies 26 Offices of chiropractors 26 Special food services 26 Sporting goods and musical instrument stores 26 Sporting goods stores 26 Coin-operated laundries and drycleaners 26 Hair 26 Hobby 26 Florists 27 Motion picture and sound recording industries 27 Janitorial services 27 Cafeterias 27 Beer 27 Social science and humanities research 27 Offices of mental health practitioners 27 Personal and laundry services 27 Other personal services 27 RV parks and recreational camps 27 Services for the elderly and persons with disabilities 27 Offices of other health practitioners 28 Offices of specialty therapists 28 Offices of dentists 28 Offices of all other health practitioners 28 Specialty food stores 28 Supermarkets and other grocery stores 28 Car washes 28 Marketing research and public opinion polling 28 Miscellaneous store retailers 28 Food and beverage stores 28 Food service contractors 28 Freestanding emergency medical centers 28 Home furnishings stores 28 Grocery stores 28 Independent artists 28 Individual and family services 28 Home health care services 29 Home centers 29 General merchandise stores 29 Death care services 29 Spectator sports 29 Social assistance 29 Veterinary services 29 Used merchandise stores 29 Offices of optometrists 29 Parking lots and garages 29 Pet and pet supplies stores 30 Other miscellaneous store retailers 30 Offices of mental health physicians 30 Office supplies and stationery stores 30 Other health and personal care stores 30 Retail trade 30 Pharmacies and drug stores 30 Traveler accommodation 30 Vocational rehabilitation services 30 Convenience stores 30 Cosmetic and beauty supply stores 30 Child and youth services 30 Child day care services 30 Accommodation 30 Gasoline stations 30 Gasoline stations with convenience stores 30 Hardware stores 30 Health and personal care stores 30 Hotels and motels 30 Membership associations and organizations 30 Jewelry 30 Labor unions and similar labor organizations 31 Lessors of nonresidential buildings 31 Furniture and home furnishings stores 31 Assisted living facilities for the elderly 31 Building material and garden supply stores 31 Building material and supplies dealers 31 Used household and office goods moving 31 Support activities for water transportation 31 Radio broadcasting 31 Services to buildings and dwellings 31 School and employee bus transportation 31 Other automotive repair and maintenance 31 Other individual and family services 31 Other professional and technical services 31 Other services 31 Other gasoline stations 32 Private service-providing 32 Photographic services 32 Ambulatory health care services 32 Carpet and upholstery cleaning services 32 All other support services 32 Airport operations 32 Convention and trade show organizers 32 Community care facilities for the elderly 32 Continuing care retirement communities 32 Education and health services 32 Human rights organizations 32 Household goods repair and maintenance 33 Health care and social assistance 33 Framing contractors 33 Grantmaking and giving services 33 Grantmaking foundations 33 Lessors of real estate 33 Marinas 33 Newspaper publishers 

industry_name LOWESTHOURS Fitness and recreational sports centers 17 Family clothing stores 18 Civic and social organizations 18 Women's clothing stores 18 Skiing facilities 19 Clothing stores 19 NULL 19 Bowling centers 20 Clothing accessories stores 21 Clothing and clothing accessories stores 21 Drinking places 21 Amusements 22 Gift 22 Snack and nonalcoholic beverage bars 22 Shoe stores 23 Other personal care services 23 Arts 23 Book stores and news dealers 24 Food services and drinking places 24 Full-service restaurants 24 Limited-service restaurants 24 Leisure and hospitality 24 Other home furnishings stores 24 Restaurants and other eating places 24 Performing arts and spectator sports 25 Performing arts companies 25 Personal care services 25 Pet care services 25 Special food services 25 Sporting goods 25 Museums 25 Office supplies 25 Golf courses and country clubs 25 Hair 25 Hobby 25 Accommodation and food services 25 Florists 25 Coin-operated laundries and drycleaners 26 Beer 26 Cafeterias 26 Car washes 26 Sporting goods and musical instrument stores 26 Sporting goods stores 26 Offices of chiropractors 26 Offices of dentists 27 Offices of mental health practitioners 27 Personal and laundry services 27 Other personal services 27 RV parks and recreational camps 27 Services for the elderly and persons with disabilities 27 Social science and humanities research 27 Spectator sports 27 Home furnishings stores 27 Janitorial services 27 Food service contractors 27 Motion picture and sound recording industries 27 Miscellaneous store retailers 28 Marketing research and public opinion polling 28 Freestanding emergency medical centers 28 General merchandise stores 28 Independent artists 28 Individual and family services 28 Grocery stores 28 Food and beverage stores 28 Specialty food stores 28 Supermarkets and other grocery stores 28 Parking lots and garages 28 Offices of optometrists 28 Offices of other health practitioners 28 Offices of all other health practitioners 28 Offices of specialty therapists 28 Offices of mental health physicians 29 Pet and pet supplies stores 29 Pharmacies and drug stores 29 Other miscellaneous store retailers 29 Social assistance 29 Services to buildings and dwellings 29 School and employee bus transportation 29 Traveler accommodation 29 Used merchandise stores 29 Veterinary services 29 Convenience stores 29 Death care services 29 Cosmetic and beauty supply stores 29 Accommodation 29 Health and personal care stores 29 Home centers 29 Home health care services 29 Hotels and motels 29 Office supplies and stationery stores 29 Membership associations and organizations 30 Lessors of nonresidential buildings 30 Hardware stores 30 Jewelry 30 Gasoline stations 30 Gasoline stations with convenience stores 30 Carpet and upholstery cleaning services 30 Child and youth services 30 Child day care services 30 Drycleaning and laundry services 30 Vocational rehabilitation services 30 Used household and office goods moving 30 Retail trade 30 Other automotive repair and maintenance 30 Other health and personal care stores 30 Other individual and family services 30 Other professional and technical services 31 Other services 31 Photographic services 31 Radio broadcasting 31 Support activities for water transportation 31 Community care facilities for the elderly 31 Building material and garden supply stores 31 Building material and supplies dealers 31 Assisted living facilities for the elderly 31 Ambulatory health care services 31 Grantmaking foundations 31 Furniture and home furnishings stores 31 Labor unions and similar labor organizations 31 Human rights organizations 31 Health care and social assistance 32 Framing contractors 32 Landscaping services 32 Lawn and garden equipment and supplies stores 32 Marinas 32 Nursing and residential care facilities 32 Nursing care facilities 32 All other support services 32 Air transportation 32 Airport operations 32 Convention and trade show organizers 32 Continuing care retirement co

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest?</span>

In [21]:
-- SELECT i.industry_code 
-- FROM LaborStatisticsDB.dbo.industry as i 
-- WHERE i.industry_code LIKE (

-- )

SELECT i.industry_name, ROUND(MAX(j.value), 0) AS 'HIGHESTPAYROLL'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE  j.series_id LIKE '%82'
GROUP BY industry_name
ORDER BY HIGHESTPAYROLL

SELECT i.industry_name, ROUND(MIN(j.value), 0) AS 'LOWESTPAYROLL'
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code
WHERE  j.series_id LIKE '%82'
GROUP BY industry_name
ORDER BY LOWESTPAYROLL

---this shows the hours, but what is the industry_name?

(501 rows affected)

(501 rows affected)

Total execution time: 00:00:00.290

industry_name HIGHESTPAYROLL Coin-operated laundries and drycleaners 10145 Clothing accessories stores 12324 Seafood product preparation and packaging 14293 Marinas 14400 Bowling centers 14549 Land subdivision 15446 Florists 16218 Crushed and broken limestone mining 16676 Carpet and upholstery cleaning services 18321 Siding contractors 18613 General rental centers 19211 Home health equipment rental 19219 RV parks and recreational camps 19300 Skiing facilities 19641 Nursery stock and florists' supplies 21665 Dental laboratories 21848 Photographic services 22545 Business to business electronic markets 22550 Human rights organizations 23871 Book stores and news dealers 24083 Computer and office machine repair 25289 Household appliance stores 25474 Landscape architectural services 26007 Wood windows and doors 26937 Cafeterias 28765 Offices of mental health physicians 29089 Independent artists 29306 Farm machinery and equipment 29362 Other building exterior contractors 31271 Wood containers and pallets 32406 Paper bags and coated and treated paper 32487 Stone mining and quarrying 32512 All other wood products 32711 Pet care services 32767 Packaging and labeling services 32871 Signs 32986 Household goods repair and maintenance 33039 Upholstered household furniture 33068 Optical goods stores 33385 Title abstract and settlement offices 33778 Electroplating 34594 Communications equipment 34623 Airport operations 35098 Reinsurance carriers 35107 Logging 35154 Gift 35331 Business service centers 36462 Masonry materials 37990 Other gasoline stations 38489 Telecommunications resellers 38497 Offices of mental health practitioners 38858 Sugar and confectionery products 38865 Pet and pet supplies stores 39571 Other services to buildings and dwellings 39644 Other health and personal care stores 40570 Social science and humanities research 40597 Beer 40708 Water 41432 Farm product raw materials 41567 Convention and trade show organizers 41922 Tile and terrazzo contractors 43083 Floor covering stores 43527 Convenience stores 43878 Glass and glazing contractors 43947 Used household and office goods moving 44313 Other ground passenger transportation 44450 Car washes 44884 Office supplies and stationery stores 45506 Soaps 46192 Other building finishing contractors 46298 Special tools 46358 Turned products and screws 46490 Flooring contractors 46604 Framing contractors 46986 Apparel 47135 Plywood and engineered wood products 47462 Cosmetic and beauty supply stores 47497 Surgical appliances and supplies 47596 Textile product mills 47716 Consumer lending 47850 Offices of chiropractors 48185 Parking lots and garages 48432 Public relations agencies 48609 Fruit and vegetable canning and drying 50125 Hobby 50161 Hardware stores 50385 Graphic design services 50776 Frozen food 50935 Travel agencies 52187 Other personal care services 53114 Metal valves 53117 Other home furnishings stores 53226 Support activities for road transportation 53268 Millwork 53782 Used merchandise stores 53794 Plastics packaging materials 54211 Turbine and power transmission equipment 54436 Motor vehicle gasoline engine and parts 54850 Fabricated structural metal products 54850 Other legal services 54883 Hardware 55054 Corrugated and solid fiber boxes 55252 Diagnostic imaging centers 55311 Jewelry 55857 Textile mills 56048 Recyclable materials 57797 Sawmills and wood preservation 57857 Museums 58405 Fruits and vegetables 58489 Offices of optometrists 59202 Women's clothing stores 59240 Commercial and service industry machinery 60064 Shoe stores 60070 Gambling industries 60138 Industrial supplies 60299 Coal mining 60667 Radio broadcasting 60751 Linen and uniform supply 61555 Sheet metal work 62178 Professional organizations 63114 Boilers 63119 Death care services 63203 Other household and institutional furniture 64413 Wood kitchen cabinets and countertops 64737 Specialty food stores 65409 Heavy machinery rental and leasing 65947 Motor vehicle metal stamping 65958 Farm supplies 67237 

industry_name LOWESTPAYROLL Coin-operated laundries and drycleaners 10079 Skiing facilities 10564 Marinas 11560 RV parks and recreational camps 11598 Seafood product preparation and packaging 11908 Clothing accessories stores 12034 NULL 12674 Crushed and broken limestone mining 13472 Bowling centers 13526 Florists 14152 Land subdivision 15362 Siding contractors 15509 Carpet and upholstery cleaning services 16302 General rental centers 16388 Nursery stock and florists' supplies 18120 Home health equipment rental 18941 Dental laboratories 21393 Photographic services 21611 Business to business electronic markets 21901 Book stores and news dealers 22603 Landscape architectural services 23066 Human rights organizations 23839 Computer and office machine repair 24901 Household appliance stores 25221 Wood windows and doors 25492 Stone mining and quarrying 26187 Independent artists 27777 Cafeterias 27827 Offices of mental health physicians 28539 Farm machinery and equipment 28965 All other wood products 30777 Other building exterior contractors 30821 Pet care services 31004 Sporting goods 31371 Household goods repair and maintenance 31373 Signs 31389 Gift 32171 Wood containers and pallets 32295 Paper bags and coated and treated paper 32467 Logging 32658 Packaging and labeling services 32790 Upholstered household furniture 33028 Optical goods stores 33198 Reinsurance carriers 33519 Title abstract and settlement offices 33694 Electroplating 34105 Communications equipment 34417 Museums 34710 Masonry materials 35070 Airport operations 35097 Other services to buildings and dwellings 35252 Business service centers 36056 Pet and pet supplies stores 37641 Telecommunications resellers 37814 Other gasoline stations 38029 Farm product raw materials 38307 Sugar and confectionery products 38342 Offices of mental health practitioners 38586 Used household and office goods moving 39863 Drycleaning and laundry services 39925 Social science and humanities research 39980 Flooring contractors 40281 Beer 40384 Convention and trade show organizers 40428 Water 40442 Other health and personal care stores 40503 Car washes 40511 Tile and terrazzo contractors 40538 Framing contractors 40978 Other building finishing contractors 41225 Convenience stores 42598 Floor covering stores 42924 Other ground passenger transportation 43462 Glass and glazing contractors 43606 Fruit and vegetable canning and drying 43892 Office supplies and stationery stores 44223 Soaps 45326 Plywood and engineered wood products 45868 Special tools 46162 Turned products and screws 46384 Textile product mills 46476 Public relations agencies 46879 Apparel 47057 Cosmetic and beauty supply stores 47107 Surgical appliances and supplies 47541 Parking lots and garages 47559 Consumer lending 47730 Offices of chiropractors 47845 Hardware stores 48986 Hobby 49296 Graphic design services 50247 Frozen food 50577 Travel agencies 51070 Other personal care services 52327 Support activities for road transportation 52446 Other home furnishings stores 52494 Millwork 52592 Metal valves 53003 Used merchandise stores 53063 Plastics packaging materials 53390 Fabricated structural metal products 53395 Turbine and power transmission equipment 53790 Jewelry 54140 Other legal services 54199 Hardware 54365 Diagnostic imaging centers 54408 Corrugated and solid fiber boxes 54492 Motor vehicle gasoline engine and parts 54753 Ready-mix concrete 54810 Textile mills 55291 Women's clothing stores 55965 Offices of optometrists 56865 Recyclable materials 57055 Sawmills and wood preservation 57110 Lawn and garden equipment and supplies stores 57987 Fruits and vegetables 58345 Gambling industries 59338 Industrial supplies 59604 Coal mining 59656 Shoe stores 59660 Commercial and service industry machinery 59905 Radio broadcasting 60605 Linen and uniform supply 61386 Professional organizations 61397 Sheet metal work 61623 Farm supplies 61663 Wood kitchen cabinets and countertops 62163 Specialty food stores 62273 Death care services

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [26]:
SELECT j.series_id, j.value, s.industry_code, i.industry_name
FROM LaborStatisticsDB.dbo.january_2017 as j 
JOIN LaborStatisticsDB.dbo.series as s 
ON j.series_id = s.series_id 
JOIN LaborStatisticsDB.dbo.industry as i 
ON i.industry_code = s.industry_code 
WHERE j.value > (
    SELECT AVG(a.value) 
    FROM LaborStatisticsDB.dbo.annual_2016 as a 
    WHERE a.series_id LIKE '%82'
)
-- only put '82' from annual_2016


(754 rows affected)

Total execution time: 00:00:14.898

series_id value industry_code industry_name CES0500000056 4239112 05000000 Total private CES0500000056 4239112 05000000 Total private CES0500000057 110301694 05000000 Total private CES0500000057 110301694 05000000 Total private CES0500000081 3412651 05000000 Total private CES0500000081 3412651 05000000 Total private CES0500000082 74498171 05000000 Total private CES0500000082 74498171 05000000 Total private CES0600000057 21779094 06000000 Goods-producing CES0600000057 21779094 06000000 Goods-producing CES0600000082 13426820 06000000 Goods-producing CES0600000082 13426820 06000000 Goods-producing CES0800000056 3432382 08000000 Private service-providing CES0800000056 3432382 08000000 Private service-providing CES0800000057 88280865 08000000 Private service-providing CES0800000057 88280865 08000000 Private service-providing CES0800000081 2818821 08000000 Private service-providing CES0800000081 2818821 08000000 Private service-providing CES0800000082 60914722 08000000 Private service-providing CES0800000082 60914722 08000000 Private service-providing CES1000000057 966707 10000000 Mining and logging CES1000000057 966707 10000000 Mining and logging CES1021000057 927250 10210000 Mining CES1021000057 927250 10210000 Mining CES2000000057 7555007 20000000 Construction CES2000000057 7555007 20000000 Construction CES2000000082 5335274 20000000 Construction CES2000000082 5335274 20000000 Construction CES2023600057 1765016 20236000 Construction of buildings CES2023600057 1765016 20236000 Construction of buildings CES2023600082 1008035 20236000 Construction of buildings CES2023600082 1008035 20236000 Construction of buildings CES2023620057 990555 20236200 Nonresidential building CES2023620057 990555 20236200 Nonresidential building CES2023700057 1221764 20237000 Heavy and civil engineering construction CES2023700057 1221764 20237000 Heavy and civil engineering construction CES2023700082 908374 20237000 Heavy and civil engineering construction CES2023700082 908374 20237000 Heavy and civil engineering construction CES2023800057 4595075 20238000 Specialty trade contractors CES2023800057 4595075 20238000 Specialty trade contractors CES2023800082 3421902 20238000 Specialty trade contractors CES2023800082 3421902 20238000 Specialty trade contractors CES2023810057 833625 20238100 Building foundation and exterior contractors CES2023810057 833625 20238100 Building foundation and exterior contractors CES2023820057 2326342 20238200 Building equipment contractors CES2023820057 2326342 20238200 Building equipment contractors CES2023820082 1696080 20238200 Building equipment contractors CES2023820082 1696080 20238200 Building equipment contractors CES2023821057 1039534 20238210 Electrical contractors CES2023821057 1039534 20238210 Electrical contractors CES2023822057 1115270 20238220 Plumbing and HVAC contractors CES2023822057 1115270 20238220 Plumbing and HVAC contractors CES3000000057 13282613 30000000 Manufacturing CES3000000057 13282613 30000000 Manufacturing CES3000000082 7491723 30000000 Manufacturing CES3000000082 7491723 30000000 Manufacturing CES3100000057 8769944 31000000 Durable goods CES3100000057 8769944 31000000 Durable goods CES3100000082 4859716 31000000 Durable goods CES3100000082 4859716 31000000 Durable goods CES3133200057 1421283 31332000 Fabricated metal products CES3133200057 1421283 31332000 Fabricated metal products CES3133200082 871298 31332000 Fabricated metal products CES3133200082 871298 31332000 Fabricated metal products CES3133300057 1251217 31333000 Machinery CES3133300057 1251217 31333000 Machinery CES3133400057 1489440 31334000 Computer and electronic products CES3133400057 1489440 31334000 Computer and electronic products CES3133600057 2130760 31336000 Transportation equipment CES3133600057 2130760 31336000 Transportation equipment CES3133600082 1257764 31336000 Transportation equipment CES3133600082 1257764 31336000 Transportation equipment CES3133600157 1050677 31336001 NULL CES3133600157 1050677 31336001 NULL CES3133640

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [33]:
SELECT j.series_id, j.value, j.year, j.period 
FROM LaborStatisticsDB.dbo.january_2017 as j 
WHERE j.series_id LIKE '%30'
UNION
SELECT a.series_id, a.value, a.year, a.period  
FROM LaborStatisticsDB.dbo.annual_2016 as a 
WHERE a.series_id LIKE '%30'



-- ON j.series_id = s.series_id 
-- JOIN LaborStatisticsDB.dbo.industry as i 
-- ON i.industry_code = s.industry_code 
-- WHERE j.value > (
--     SELECT AVG(a.value) 
--     FROM LaborStatisticsDB.dbo.annual_2016 as a 
--     WHERE a.series_id LIKE '%82'

-- data_type_code = 6


(1653 rows affected)

Total execution time: 00:00:07.942

series_id value year period CES0500000030 733.49 2017 M01 CES0600000030 939.14 2017 M01 CES0800000030 698 2017 M01 CES1000000030 1253.53 2017 M01 CES1011330030 861.42 2017 M01 CES1021000030 1289.27 2017 M01 CES1021100030 1483.68 2017 M01 CES1021200030 1230.87 2017 M01 CES1021210030 1444.13 2017 M01 CES1021231030 1022.35 2017 M01 CES1021231230 1011.27 2017 M01 CES1021231930 1017.68 2017 M01 CES1021300030 1225.07 2017 M01 CES1021311230 1186.18 2017 M01 CES2000000030 1034.77 2017 M01 CES2023600030 991.38 2017 M01 CES2023610030 847.96 2017 M01 CES2023611530 854.47 2017 M01 CES2023611830 795.62 2017 M01 CES2023620030 1130.82 2017 M01 CES2023621030 1214.1 2017 M01 CES2023622030 1108.72 2017 M01 CES2023700030 1237.72 2017 M01 CES2023710030 1248.82 2017 M01 CES2023711030 1051.52 2017 M01 CES2023712030 1469.52 2017 M01 CES2023713030 1324.77 2017 M01 CES2023720030 1126.88 2017 M01 CES2023730030 1148.86 2017 M01 CES2023790030 1443.57 2017 M01 CES2023800030 1008.14 2017 M01 CES2023810030 898.56 2017 M01 CES2023811030 966.01 2017 M01 CES2023812030 1201.32 2017 M01 CES2023813030 704.37 2017 M01 CES2023814030 847.88 2017 M01 CES2023815030 955.15 2017 M01 CES2023816030 820.8 2017 M01 CES2023817030 716.04 2017 M01 CES2023819030 888.38 2017 M01 CES2023820030 1088.98 2017 M01 CES2023821030 1150.9 2017 M01 CES2023822030 1015.74 2017 M01 CES2023829030 1225.7 2017 M01 CES2023830030 931.5 2017 M01 CES2023831030 1042.34 2017 M01 CES2023832030 804.46 2017 M01 CES2023833030 945.45 2017 M01 CES2023834030 1082.4 2017 M01 CES2023835030 852.98 2017 M01 CES2023839030 865.37 2017 M01 CES2023890030 988.4 2017 M01 CES2023891030 1000.3 2017 M01 CES2023899030 967.66 2017 M01 CES3000000030 864.4 2017 M01 CES3100000030 915.37 2017 M01 CES3132100030 720.33 2017 M01 CES3132110030 775.54 2017 M01 CES3132120030 760.49 2017 M01 CES3132190030 684.6 2017 M01 CES3132191030 728.75 2017 M01 CES3132191130 778.27 2017 M01 CES3132191830 707.67 2017 M01 CES3132192030 652.19 2017 M01 CES3132199030 645.22 2017 M01 CES3132700030 855.82 2017 M01 CES3132730030 894.96 2017 M01 CES3132732030 924.32 2017 M01 CES3132739030 869.28 2017 M01 CES3133100030 982.1 2017 M01 CES3133150030 825.1 2017 M01 CES3133200030 839.47 2017 M01 CES3133210030 904.24 2017 M01 CES3133230030 796.05 2017 M01 CES3133231030 837.34 2017 M01 CES3133231230 847.76 2017 M01 CES3133232030 765.61 2017 M01 CES3133232230 768.57 2017 M01 CES3133240030 993.98 2017 M01 CES3133270030 865.58 2017 M01 CES3133271030 878.64 2017 M01 CES3133272030 811.49 2017 M01 CES3133280030 729.11 2017 M01 CES3133281230 749.89 2017 M01 CES3133281330 706 2017 M01 CES3133290030 888.68 2017 M01 CES3133291030 928.4 2017 M01 CES3133299030 864.84 2017 M01 CES3133300030 937.75 2017 M01 CES3133310030 880.65 2017 M01 CES3133311130 827.04 2017 M01 CES3133320030 978.32 2017 M01 CES3133330030 1089.91 2017 M01 CES3133340030 813.75 2017 M01 CES3133350030 979.97 2017 M01 CES3133351430 1023.37 2017 M01 CES3133360030 1046.32 2017 M01 CES3133390030 926.19 2017 M01 CES3133400030 1014.34 2017 M01 CES3133420030 792.42 2017 M01 CES3133440030 899.7 2017 M01 CES3133441330 1329.12 2017 M01 CES3133441930 755.6 2017 M01 CES3133450030 1187.62 2017 M01 CES3133500030 836.74 2017 M01 CES3133530030 855.83 2017 M01 CES3133590030 876.65 2017 M01 CES3133600030 1098.68 2017 M01 CES3133600130 940.34 2017 M01 CES3133610030 1309.25 2017 M01 CES3133620030 723.77 2017 M01 CES3133630030 869.09 2017 M01 CES3133631030 1025.36 2017 M01 CES3133635030 1206.65 2017 M01 CES3133637030 974.38 2017 M01 CES3133639030 785.37 2017 M01 CES3133640030 1592.97 2017 M01 CES3133641130 1663.59 2017 M01 CES3133700030 704.3 2017 M01 CES3133710030 685.41 2017 M01 CES3133711030 744.98 2017 M01 CES3133712030 640.85 2017 M01 CES3133712130 643.4 2017 M01 CES3133900030 760.73 2017 M01 CES3133910030 828.24 2017 M01 CES3133911330 789.39 2017 M01 CES3133911630 745.47 2017 M01 CES3133990030 679.14 2017 M01 CES3133995030 706.32 2017 M01 CES3133999030 689.48 2017 M01 CES3200000030 784.29 2017 M01 CES3231100030 694.3 20

In [53]:
SELECT AVG(j.value) AS 'JANUARY 2017'
FROM LaborStatisticsDB.dbo.january_2017 as j 
WHERE j.series_id LIKE '%30'
-- JOIN 

SELECT AVG(a.value) AS 'ANNUAL 2016'
FROM LaborStatisticsDB.dbo.annual_2016 as a 
-- ON a.series_id, a.value, a.year, a.period 
WHERE a.series_id LIKE '%30'

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.188

JANUARY 2017 808.5303811252254

ANNUAL 2016 797.1962613430123

In [72]:
SELECT SUM(j.value) as 'JANUARY FARE'
FROM LaborStatisticsDB.dbo.january_2017 as j 
WHERE j.period = 'M01' AND j.series_id LIKE '%30'


SELECT SUM(a.value) as 'ANNUAL FARE'
FROM LaborStatisticsDB.dbo.annual_2016 as a
WHERE a.period = 'M13' AND a.series_id LIKE '%30'

(1 row affected)

(1 row affected)

Total execution time: 00:00:00.224

JANUARY FARE 1782000.9599999967

ANNUAL FARE 878510.2799999996

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

During Period 'M01' or January, is when the production and nonsupervisory employees fare better as seen with the following code: 

  

<span style="color: #0000ff;">SELECT</span> <span style="color: #795e26;">SUM</span>(j.value) <span style="color: #0000ff;">as</span> <span style="color: #a31515;">'JANUARY FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.january\_2017 <span style="color: #0000ff;">as</span> j

<span style="color: #0000ff;">WHERE</span> j.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M01'</span> <span style="color: #0000ff;">AND</span> j.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span>

  
  

<span style="color: #0000ff;">SELECT</span> <span style="color: #795e26;">SUM</span>(a.value) <span style="color: #0000ff;">as</span> <span style="color: #a31515;">'ANNUAL FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.annual\_2016 <span style="color: #0000ff;">as</span> a

<span style="color: #0000ff;">WHERE</span> a.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M13'</span> <span style="color: #0000ff;">AND</span> a.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span><span style="color: rgb(0, 0, 255);">SELECT</span> <span style="color: rgb(121, 94, 38);">SUM</span>(j.value) <span style="color: rgb(0, 0, 255);">as</span> <span style="color: rgb(163, 21, 21);">'JANUARY FARE'</span>

<span style="color: #0000ff;">FROM</span> LaborStatisticsDB.dbo.january\_2017 <span style="color: #0000ff;">as</span> j

<span style="color: #0000ff;">WHERE</span> j.period <span style="color: #000000;">=</span> <span style="color: #a31515;">'M01'</span> <span style="color: #0000ff;">AND</span> j.series\_id <span style="color: #0000ff;">LIKE</span> <span style="color: #a31515;">'%30'</span>

  
  

which shows that January's fare was 1782000.96 and the 'M13' period totaled 878510.28

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better? The industries with the highest pay include "Total Private", "Private Service-Providing" and "Professional and business services."</span>

Double click here to add your answer.

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

<span style="font-size: 13.3333px; white-space: pre-wrap;">It would have been nice to have the foreign keys and primary keys listed out for us as well as a table/spreadsheet with the January_2017 and Annual_2016 infomation.</span>